src/parser/parse.ts

```diff
@@ -284,11 +273,29 @@ export class Parser<T extends ParserInput, TFlags extends OutputFlags<T['flags']
 
       if (!(k in flags) && flag.default !== undefined) {
         this.metaData.flags[k] = {...this.metaData.flags[k], setFromDefault: true}
-        const defaultValue = (typeof flag.default === 'function' ? await flag.default({options: flag, flags, ...this.context}) : flag.default)
+        const defaultValue = (typeof flag.default === 'function' ? await flag.default({
+          options: flag,
+          flags, ...this.context,
+        }) : flag.default)
         flags[k] = defaultValue
       }
     }
 
+    for (const k of Object.keys(this.input.flags)) {
+      if ((k in flags) && Reflect.has(this.input.flags[k], 'defaultHelp')) {
```

Comment:
could this be done in the previous for loop? both iterate over the keys of `this.input.flags`
https://github.com/oclif/core/blob/main/src/parser/parse.ts#L268-L290